build-id-versioning/starter/main.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"github.com/pborman/uuid"
+
+	"go.temporal.io/sdk/client"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Get task queue name from CLI arg
+	taskQueue := os.Args[1]
+	if taskQueue == "" {
+		log.Fatalln("Must provide task queue name as first and only argument")
+	}
+	log.Println("Using task queue", taskQueue)
+
+	// The client is a heavyweight object that should be created once per process.
+	c, err := client.Dial(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+
+	// First, let's make the task queue use the build id versioning feature by adding an initial
+	// default version to the queue:
+	err = c.UpdateWorkerBuildIdCompatibility(ctx, &client.UpdateWorkerBuildIdCompatibilityOptions{
+		TaskQueue: taskQueue,
+		Operation: &client.BuildIDOpAddNewIDInNewDefaultSet{
+			BuildID: "1.0",
+		},
+	})
+	if err != nil {
+		log.Fatalln("Unable to update worker build id compatibility", err)
+	}
+
+	firstWorkflowID := "build-id-versioning-first_" + uuid.New()
+	firstWorkflowOptions := client.StartWorkflowOptions{
+		ID:                       firstWorkflowID,
+		TaskQueue:                taskQueue,
+		WorkflowExecutionTimeout: 5 * time.Minute,
+	}
+	firstExecution, err := c.ExecuteWorkflow(ctx, firstWorkflowOptions, "SampleChangingWorkflow")
+	if err != nil {
+		log.Fatalln("Unable to start workflow", err)
+	}
+	log.Println("Started first workflow",
+		"WorkflowID", firstExecution.GetID(), "RunID", firstExecution.GetRunID())
+
+	// Signal this workflow a few times to drive it
+	for i := 0; i < 3; i++ {
+		err = c.SignalWorkflow(ctx, firstExecution.GetID(), firstExecution.GetRunID(),
+			"do-next-signal", "do-activity")
+		if err != nil {
+			log.Fatalln("Unable to signal workflow", err)
+		}
+	}
+
+	// Give a chance for these signals to be processed by the 1.0 worker
+	time.Sleep(5 * time.Second)
+
+	// Now, let's update the task queue with a new compatible version:
+	err = c.UpdateWorkerBuildIdCompatibility(ctx, &client.UpdateWorkerBuildIdCompatibilityOptions{
+		TaskQueue: taskQueue,
+		Operation: &client.BuildIDOpAddNewCompatibleVersion{
+			BuildID:                   "1.1",
+			ExistingCompatibleBuildId: "1.0",
```

Comment:
We missed this inconsistency in the Go SDK.

Reply:
We can change it if we want, not the most noticeable